addon/index.js

```diff
@@ -77,6 +75,51 @@ function updatePrefs() {
   });
 }
 
+// Listen for metrics events from experiments.
+const metrics = {
+  isInitialized: false,
+  init: function() {
+    if (this.isInitialized) {
+      return;
+    }
+    this.isInitialized = true;
+    // Note: the observer service holds a strong reference to this observer,
+    // so we must detach it on shutdown / uninstall by calling destroy().
+    Services.obs.addObserver(metrics, 'idea-town::send-metric', false);
+  },
+  destroy: function() {
+    Services.obs.removeObserver(metrics, 'idea-town::send-metric', false);
+    this.isInitialized = false;
+  },
+  // The metrics object format is { key, value, addonName }, where
+  //   'key' is the name of the event,
+  //   'value' is the value of the event (can be any JSON-serializable object),
+  //   'addonName' is the name of the experiment sending the data.
+  observe: function() {
+    // The nsIObserverService sends non-useful positional arguments; the third
+    // is the only one we need.
+    const data = arguments[2];
+
+    let d;
+    try {
+      d = JSON.parse(data);
+    } catch (ex) {
+      console.error('Idea Town metrics error: cannot process event, ' +
+                    ' JSON.parse failed: ', ex);
```

Comment:
@lmorchard Thoughts on the best way to notify individual addons if their metrics packets are malformed? Right now I've got these `console.error` lines, because it seemed weird to actually throw just because a fire-and-forget packet looked wrong. OTOH the eslint config complains about the use of console.